web_poet/serialization/functions.py

```diff
@@ -0,0 +1,149 @@
+import json
+from types import NoneType
+from typing import Any, Dict, Generic, Type, Union
+
+import attrs
+
+from .. import (
+    HttpResponse,
+    HttpResponseBody,
+    HttpResponseHeaders,
+    Injectable,
+    ResponseUrl,
+)
+from ..page_inputs.url import _Url
+from .api import SerializedData, deserialize, register_serialization, serialize
+
+
+def _serialize_dict(o: dict) -> SerializedData:
+    return {"json": json.dumps(o).encode()}
+
+
+def _deserialize_dict(t: Type[dict], data: SerializedData) -> dict:
+    return t(json.loads(data["json"].decode()))
+
+
+register_serialization(_serialize_dict, _deserialize_dict)
+
+
+def _serialize_Injectable(o: Injectable) -> SerializedData:
+    # None is Injectable
+    if o is None:
+        return {}
+
+    assert attrs.has(o)  # FIXME
+    deps = attrs.asdict(o, recurse=False)
+    return {dep_name: serialize(dep) for dep_name, dep in deps.items()}
+
+
+def _deserialize_Injectable(t: Type[Injectable], data: SerializedData) -> Injectable:
+    def _strip_Optional(t: Type) -> Type:
```

Comment:
This is special support for Optional but other non-trivial field types could need additional support too.